Least squares cholesky

Recktenwald opened the pull request:
This is adding a least squares algorithm based on the Cholesky Decomposition, as shown in Issue 162 https://github.com/fslaborg/FSharp.Stats/issues/162 

As a side note: My autoformatter didn't like lines like `let LU A = ...` and would automatically change it to `let LUA = ...` so I added some parentheses and type annotations at such places to avoid that, and because I think it is good style to be explicit.

Commit:
Add implementations for the least squares algorithm using Cholesky Decomposition

nhirschey commented:
Nice idea for a contribution. I think you could make it easier for the maintainers to evaluate your changes if you 

1. Removed all your auto-formatting. It adds a lot of noise and makes it hard to separate the real changes from the formatting changes.
2. Added a test to verify that your implementation is correct.

bvenn commented:
This is a nice one @Recktenwald 🚀 
As @nhirschey already mentioned if would be great, if you could open a RP without the autoformatted changes. It makes it hard to identify the points of your contribution regarding the Cholesky decomposition. I'm currently looking into fantomas to format the FSharp.Stats codebase.

Additionally could you please provide an description for `leastSquaresCholesky` and a test.

Thanks for reporting the missing template. I'm going to fix it.

Recktenwald commented:
I rebased and removed two of the commits that mainly included formatting. There is still quite a bit noise left but it is more localized. I'm too inexperienced with git to do much about those. At this point, it would probably be more painless to just delete my repo, delete this PR, make a new fork, and paste the actual content in again. Do you think that's necessary? Would that break something?

I have not added a test, yet. I'd wait with that until you give me the green light with respect to the remaining noisy formatting changes.

bvenn commented:
Probably that would be the easiest option for you to create a clean branch, add your changes, and file a new PR.
